src/v/utils/mutex.h

```diff
@@ -48,10 +52,24 @@ class basic_mutex {
         return ss::with_semaphore(_sem, 1, std::forward<Func>(func));
     }
 
+    template<typename Func>
+    auto with(Func&& func, duration timeout) noexcept {
+        return ss::with_semaphore(_sem, 1, timeout, std::forward<Func>(func));
+    }
+
+    template<typename Func>
+    auto with(Func&& func, time_point timeout) noexcept {
+        return ss::get_units(_sem, 1, timeout)
+          .then([func = std::forward<Func>(func)](auto units) mutable {
+              return ss::futurize_invoke(std::forward<Func>(func))
+                .finally([units = std::move(units)] {});
+          });
```

Comment:
should we create a file with helper methods pending upstreaming into seastar? this is the time_point variant of `ss::with_semaphore`, right? i don't see the `time_point` variant being used in this patch series? is it being used in another PR?

Reply:
yes, it is intended to be used in other places

Reply:
this is the time_point variant of ss::with_semaphore, right? we should figure out some mechanism to push these back up into seastar as contributions.